src/corehost/cli/runtime_config.cpp

```diff
@@ -201,14 +280,60 @@ bool runtime_config_t::ensure_parsed()
         trace::verbose(_X("UTF-8 BOM skipped while reading [%s]"), m_path.c_str());
     }
 
+    bool rc = true;
     try
     {
         const auto root = json_value::parse(file);
         const auto& json = root.as_object();
         const auto iter = json.find(_X("runtimeOptions"));
         if (iter != json.end())
         {
-            parse_opts(iter->second);
+            rc = parse_opts(iter->second);
+
+            if (rc)
+            {
+                if (app_config == nullptr)
+                {
+                    // For the app, populate the additionalFrameworks section that is used to override framework values
+                    // and remember the settings so we can apply later while each framework's runtimeconfig is being read.
+                    const auto& opts_obj = iter->second.as_object();
+                    const auto iter = opts_obj.find(_X("additionalFrameworks"));
+                    if (iter != opts_obj.end())
+                    {
+                        const auto& additional_frameworks = iter->second.as_array();
+                        for (const auto& fx : additional_frameworks)
+                        {
+                            runtime_config_t fx_overrides;
+                            const auto& fx_obj = fx.as_object();
+                            fx_overrides.m_fx_name = fx_obj.at(_X("name")).as_string();
+                            if (fx_overrides.m_fx_name.length() == 0)
+                            {
+                                trace::verbose(_X("No framework name in additionalFrameworks section."));
+                                rc = false;
+                                break;
+                            }
+
+                            rc = fx_overrides.parse_framework(fx_obj);
+                            if (!rc)
+                            {
+                                break;
+                            }
+
+                            m_additional_frameworks[fx_overrides.m_fx_name] = fx_overrides.m_fx;
+                        }
+                    }
+
+                    // Use the framework specified if it exists in the additionalFramework
+                    app_config = this;
```

Comment:
Review note: by falling through to the code below, this means the "additionalFrameworks" section can be used to override the current (first) framework the app specifies. We could prevent that from occurring if we want to limit redundant cases, as now the "framework" section can also be used to specify the other settings besides 'version' - applyPatches, etc